### PR TITLE
Upgrading the mockito library to Mockito 2, which allows us to Mock t…

### PR DESCRIPTION
**Pull request by RobertKeazor**
…he Unmockable ().This is important so we don't have to declare our viewModels as open , just so we can stub it out

### CODE REVIEW

**Comment by RobertKeazor** [comment]
@ujavid

File: app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt [hunk 1]
```diff
@@ -1,20 +1,32 @@
...
 @RunWith(JUnitPlatform::class)
 class MainViewModelSpek : Spek ({
-    val viewModel = MainViewModel()
+    //TODO This is not a actual ViewModelSpek. Replace with actual Spek, when real app code is introduced
+    describe("mocking with mockito in spek") {
 
-    context("Testing our hello world") {
+        given("mocking Iterable<T> and Iterator<T>") {
+            val mockedIterable = Mockito.mock(Iterable::class.java)
+            val mockedIterator = Mockito.mock(Iterator::class.java)
```
Comment (ujavid) [inline app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt:20]: You got a duplicate line here

File: app/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker [hunk 2]
```diff
@@ -0,0 +1 @@
+mock-maker-inline
```
Comment (ujavid) [inline app/src/test/resources/mockito-extensions/org.mockito.plugins.MockMaker:1]: :space_invader:

File: app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt (hunk 1 above)
Comment (RobertKeazor) [reply app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt:20]: duplicated line?

Comment (ujavid) [reply app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt:20]: With the one right above it

Comment (ujavid) [reply app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt:20]: Oh lol NVM just looked like it was

**Review by ujavid — APPROVED** [review]

File: app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt (hunk 1 above)
Comment (RobertKeazor) [reply app/src/test/java/com/example/kotlin/movieapp/ui/main/MainViewModelSpek.kt:20]: lol

**Comment by RobertKeazor** [comment]
Lol now to figure out how this dependency injection will work  .

**Pull request merged**